doc/src/modules/polys/basics.rst

```diff
@@ -14,6 +14,319 @@ polynomials within SymPy. All code examples assume::
+    >>> e.as_poly(extension=True)
+    Poly(x + sqrt(2), x, domain='QQ<sqrt(2)>')
+
+With the default setting ``extension=False`` `x` and `\sqrt 2` are
+incorrectly considered algebraically independent. With coefficients in
+the extension field `\mathbb{Q}(\sqrt 2)` the square root is treated
+properly as an algebraic number.
+
+Divisibility
+------------
+
+The fourth rational operation, division, or inverted multiplication,
+is not generally possible in rings. If `a` and `b` are two elements
+of a ring `A`, then there may exist a third element `q` in `A` such
+that `a = bq`. In fact, there may exist several such elements.
+
+If also `a = bq'` for some `q'` in `A`, then `b(q - q') = 0`. Hence
+either `b` or `q - q'` is zero, or they are both *zero divisors*,
+nonzero elements whose product is zero.
+
+Integral domains
+````````````````
+Commutative rings with no zero divisors are called *integral domains*.
+Most of the commonly encountered rings, the ring of integers, fields,
+and polynomial rings over integral domains are integral domains.
+
+Assume now that `A` is an integral domain, and consider the set
+`P` of its nonzero elements, which is closed under multiplication.
+If `a` and `b` are in `P`, and there exists an element `q` in `P`
+such that `a = bq`, then `q` is unique and called the *quotient*,
+`a/b`, of `a`  by `b`. Moreover, it is said that
+
+- `a` is *divisible* by `b`,
+
+- `b` is a *divisor* of `a`,
+
+- `a` is a *multiple* of `b`,
+
+- `b` is a *factor* of `a`.
+
+An element `a` of `P` is a divisor of `1` if and only if it
+is *invertible* in `A`, with the inverse `a^{-1} = 1/a`. Such
+elements are called *units*. The units of the ring
+of integers are `1` and `-1`. The invertible elements
+in a polynomial ring over a field are the nonzero constant polynomials.
+
+If two elements of `P`, `a` and `b`, are divisible by each other, then
+the quotient `a/b` is invertible with inverse `b/a`, or equivalently,
+`b = ua` where `u` is a unit. Such elements are said to be *associated*
+with, or *associates* of, each other. The associates of an integer
+`n` are `n` and `-n`. In a polynomial ring over a field the associates
+of a polynomial are its constant multiples.
+
+Each element of `P` is divisible by its associates and the units. An
+element is *irreducible* if it has no other divisors and is not a unit.
+The irreducible elements in the ring of integers are the prime numbers
+`p` and their opposites `-p`. In a field, every nonzero element is
+invertible and there are no irreducible elements.
+
+Factorial domains
+`````````````````
+In the ring of integers, each nonzero element can be represented as
+a product irreducible elements and optionally a unit `\pm 1`. Moreover,
+any two such products have the same number of irreducible factors
+which are associated with each other in a suitable order.
+Integral domains having this property are called *factorial*,
+or *unique factorization domains*. In addition to the ring of
```

Comment:
I'm not acquainted well with the terminology but I wonder if it should read "*factorial*, or *unique factorization*, domains".